src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowManager.java

```diff
@@ -55,11 +54,12 @@ public Future<Void> updatePiece(Piece pieceToUpdate, boolean createItem, boolean
     pieceStorageService.getPieceById(pieceToUpdate.getId(), requestContext)
       .onSuccess(holder::withPieceFromStorage)
       .compose(aHolder -> basePieceFlowHolderBuilder.updateHolderWithOrderInformation(holder, requestContext))
+      .compose(aHolder -> basePieceFlowHolderBuilder.updateHolderWithTitleInformation(holder, requestContext))
       .map(v -> {
         defaultPieceFlowsValidator.isPieceRequestValid(pieceToUpdate, holder.getOriginPoLine(), createItem);
         return null;
       })
-      .compose(purchaseOrder -> protectionService.isOperationRestricted(holder.getOriginPurchaseOrder().getAcqUnitIds(), UPDATE, requestContext))
+      .compose(title -> protectionService.isOperationRestricted(holder.getTitle().getAcqUnitIds(), UPDATE, requestContext))
```

Comment:
Forgot to mention in the ticket, PieceCreateFlowManager and PieceDeleteFlowManager also should be updated with similar logic to get acq units from title instead of order

Reply:
added